Lib/ntpath.py

```diff
@@ -41,14 +42,36 @@ def _get_bothseps(path):
 # Other normalizations (such as optimizing '../' away) are not done
 # (this is done by normpath).
 
-def normcase(s):
-    """Normalize case of pathname.
+try:
+    from _winapi import (
+        LCMapStringEx as _LCMapStringEx,
+        LOCALE_NAME_INVARIANT as _LOCALE_NAME_INVARIANT,
+        LCMAP_LOWERCASE as _LCMAP_LOWERCASE)
+
+    def normcase(s):
+        """Normalize case of pathname.
+
+        Makes all characters lowercase and all slashes into backslashes.
+        """
+        s = os.fspath(s)
+        if not s:
+            return s
+        if isinstance(s, bytes):
+            s = os.fsdecode(s).replace('/', '\\')
+            return os.fsencode(_LCMapStringEx(_LOCALE_NAME_INVARIANT,
+                                              _LCMAP_LOWERCASE, s))
```

Comment:
The "surrogateescape" error handler is compatible with the "surrogatepass" filesystem error handler (Windows filesystems allow lone surrogate codes, so they have to be passed without an exception), and the surrogate escape codes are compatible with the use of `LCMapStringEx()` here. So I suggest following Steve's advice to use "surrogateescape" here:

```suggestion
            encoding = sys.getfilesystemencoding()
            s = s.decode(encoding, 'surrogateescape').replace('/', '\\')
            s = _LCMapStringEx(_LOCALE_NAME_INVARIANT,
                               _LCMAP_LOWERCASE, s)
            return s.encode(encoding, 'surrogateescape')
```

Reply:
This shouldn't be needed in the fallback code when `ntpath` is used in POSIX, because the error handler for `os.fsdecode()` is "surrogateescape" in POSIX.